sonar-db/src/main/java/org/sonar/db/profiling/ProfilingPreparedStatementHandler.java

```diff
@@ -19,43 +19,44 @@
  */
 package org.sonar.db.profiling;
 
-import com.google.common.collect.Lists;
 import java.lang.reflect.InvocationHandler;
 import java.lang.reflect.Method;
 import java.sql.PreparedStatement;
-import java.util.List;
 import org.apache.commons.lang.StringUtils;
 import org.sonar.api.utils.log.Profiler;
 
+import static org.sonar.db.profiling.SqlLogFormatter.FORMATTER;
+
 class ProfilingPreparedStatementHandler implements InvocationHandler {
 
   private final PreparedStatement statement;
-  private final List<Object> arguments;
   private final String sql;
+  private final Object[] sqlParams;
 
   ProfilingPreparedStatementHandler(PreparedStatement statement, String sql) {
     this.statement = statement;
     this.sql = sql;
-    this.arguments = Lists.newArrayList();
-    for (int argCount = 0; argCount < StringUtils.countMatches(sql, "?"); argCount++) {
-      arguments.add("!");
-    }
+    int argCount = StringUtils.countMatches(sql, "?");
```

Comment:
we should have our own implementation of countMatches which take a char as an argument
this method from StringUtils is based on `String.indexOf(String, int)` and I bet it could be lightning faster to just loop over the char array of the string